lib/std/os/linux.zig

```diff
@@ -5597,3 +5600,72 @@ pub const AUDIT = struct {
         }
     };
 };
+
+pub const IoctlDirection = enum {
+    none,
+    read,
+    write,
+    read_write,
+};
+
+/// Returns a type-safe ioctl wrapper function.
+///
+/// Linux encodes the size of the ioctl argument and its IO direction within the request number.
+/// `ioctlFunc` returns a function with the same argument type that was used to create the
+/// request number.  By preserving this relationship, the kernel will always have the correct
+/// IO direction and argument size, meaning it should never result in out-of-bounds memory access.
+pub fn ioctlFunc(
+    comptime dir: IoctlDirection,
+    comptime io_type: u8,
+    comptime nr: u8,
+    comptime Arg: type,
+) ioctlInfo(dir, io_type, nr, Arg).Fn {
```

Comment:
This should return a function literal instead of a function pointer.

Reply:
done